version_info.go

```diff
@@ -0,0 +1,306 @@
+func (f Info) FileVersion() string {
+	return f.GetProperty("FileVersion")
+}
+
+// InternalName returns InternalName property.
+func (f Info) InternalName() string {
+	return f.GetProperty("InternalName")
+}
+
+// LegalCopyright returns LegalCopyright property.
+func (f Info) LegalCopyright() string {
+	return f.GetProperty("LegalCopyright")
+}
+
+// OriginalFilename returns OriginalFilename property.
+func (f Info) OriginalFilename() string {
+	return f.GetProperty("OriginalFilename")
+}
+
+// ProductName returns ProductName property.
+func (f Info) ProductName() string {
+	return f.GetProperty("ProductName")
+}
+
+// ProductVersion returns ProductVersion property.
+func (f Info) ProductVersion() string {
+	return f.GetProperty("ProductVersion")
+}
+
+// Comments returns Comments property.
+func (f Info) Comments() string {
+	return f.GetProperty("Comments")
+}
+
+// FileVeLegalTrademarksrsion returns FileVeLegalTrademarksrsion property.
+func (f Info) FileVeLegalTrademarksrsion() string {
+	return f.GetProperty("LegalTrademarks")
+}
+
+// PrivateBuild returns PrivateBuild property.
+func (f Info) PrivateBuild() string {
+	return f.GetProperty("PrivateBuild")
+}
+
+// SpecialBuild returns SpecialBuild property.
+func (f Info) SpecialBuild() string {
+	return f.GetProperty("SpecialBuild")
+}
+
+// New creates Info instance with default locale.
+func New(path string) (Info, error) {
+	info, err := newWithoutLocale(path)
+	if err != nil {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo; %s", err)
+	}
+
+	pageID := info.GetPageID()
+	info.pageID = pageID
+
+	return info, nil
+}
+
+// NewWithLocale creates Info instance with user-defined locale.
+func NewWithLocale(path string, pageID Locale) (Info, error) {
+	info, err := newWithoutLocale(path)
+	if err != nil {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo; %s", err)
+	}
+
+	info.pageID = pageID
+
+	return info, nil
+}
+
+func newWithoutLocale(path string) (Info, error) {
+	pathPtr, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return Info{}, xerrors.Errorf("failed to convert image path to utf16; %s", err)
+	}
+	size, _, err := getFileVersionInfoSizeProc.Call(
+		uintptr(unsafe.Pointer(pathPtr)),
+		0,
+	)
+	if size == 0 {
+		return Info{}, xerrors.Errorf("failed to get memory size for VersionInfo slice; %s", err)
+	}
+	info := make([]byte, size)
+	ret, _, err := getFileVersionInfoProc.Call(
+		uintptr(unsafe.Pointer(pathPtr)),
+		0,
+		uintptr(len(info)),
+		uintptr(unsafe.Pointer(&info[0])),
+	)
+	if ret == 0 {
+		return Info{}, xerrors.Errorf("failed to get VersionInfo from windows; %s", err)
+	}
+
+	vi := Info{data: info}
+
+	fixedInfo := vi.GetFixedInfo()
+	vi.FixedFileInfo = fixedInfo
+
+	return vi, nil
+}
+
+// GetPageID gets pageID filed in structure.
+// It tries to get pageID from VersionInfo data.
+// If getting ends with fail, it returns default pageID
+// 040904E4 // US English + CP_USASCII.
+func (f Info) GetPageID() Locale {
```

Comment:
Here we should have something like `Locales() []Locale` to allow user to choose the right one.

\+ I think we need to make `f.pageID` a public property to check or change it before quering properties